src/static/helpers/slasHelper.ts

```diff
@@ -147,6 +150,55 @@ export async function authorize(
   return {url: redirectUrlString, ...getCodeAndUsidFromUrl(redirectUrlString)};
 }
 
+/**
+ * A single function to execute the ShopperLogin Private Client Guest Login as described in the [API documentation](https://developer.salesforce.com/docs/commerce/commerce-api/guide/slas-private-client.html).
+ *
+ * @param slasClient - a configured instance of the ShopperLogin SDK client
+ * @param credentials - client secret used for authentication
+ * @param credentials.clientSecret - secret associated with client ID
+ * @param parameters - parameters to pass in the API calls.
+ * @param parameters.redirectURI - Per OAuth standard, a valid app route. Must be listed in your SLAS configuration. On server, this will not be actually called. On browser, this will be called, but ignored.
+ * @param parameters.usid? - Unique Shopper Identifier to enable personalization.
+ * @returns TokenResponse
+ */
+export async function loginGuestUserPrivate(
+  slasClient: ShopperLogin<{
+    shortCode: string;
+    organizationId: string;
+    clientId: string;
+    siteId: string;
+  }>,
+  parameters: {
+    redirectURI: string;
+    usid?: string;
+  },
+  credentials: {
+    clientSecret: string;
+  }
+): Promise<TokenResponse> {
+  /* istanbul ignore next */
+  if (isBrowser) {
+    // we intentionally have warning here
+    // eslint-disable-next-line
+    console.warn(warningSlasSecretMsg);
+  }
+  const authorization = `Basic ${stringToBase64(
+    `${slasClient.clientConfig.parameters.clientId}:${credentials.clientSecret}`
+  )}`;
+
+  const options = {
+    headers: {
+      Authorization: authorization,
+    },
+    body: {
+      grant_type: 'client_credentials',
+      ...(parameters.usid && {usid: parameters.usid}),
```

Comment:
We accept `redirectURI` but never pass it onto the request, it should be a part of the body: https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-login?meta=getAccessToken

Reply:
I read the private flow again, we don't need to pass redirectURI for private flow. uuisd and grant_type should suffice https://developer.salesforce.com/docs/commerce/commerce-api/guide/slas-private-client.html